conan/tools/premake/premakedeps.py

```diff
@@ -37,74 +109,217 @@ def __init__(self, deps_cpp_info):
+        self.architecture = conanfile.settings.arch
+
     def generate(self):
+        """
+        Generates ``conan_<pkg>_vars_<config>.premake5.lua``, ``conan_<pkg>_<config>.premake5.lua``,
+        and ``conan_<pkg>.premake5.lua`` files into the ``conanfile.generators_folder``.
+        """
+
         check_duplicated_generator(self, self._conanfile)
         # Current directory is the generators_folder
         generator_files = self.content
         for generator_file, content in generator_files.items():
             save(generator_file, content)
 
-    def _get_cpp_info(self):
-        ret = CppInfo()
-        for dep in self._conanfile.dependencies.host.values():
-            dep_cppinfo = dep.cpp_info.copy()
-            dep_cppinfo.set_relative_base_folder(dep.package_folder)
-            # In case we have components, aggregate them, we do not support isolated "targets"
-            dep_cppinfo.aggregate_components()
-            ret.merge(dep_cppinfo)
-        return ret
+    def _config_suffix(self):
+        props = [("Configuration", self.configuration),
+                 ("Platform", self.architecture)]
+        name = "".join("_%s" % v for _, v in props)
+        return name.lower()
+    
+    def _output_lua_file(self, filename, content):
+        self.output_files[filename] = "\n".join(["#!lua", *content])
+
+    def _indent_string(self, string, indent=1):
+        return "\n".join([
+            f"{self.tab * indent}{line}" for line in list(filter(None, string.splitlines()))
+        ])
+
+    def _premake_filtered(self, content, configuration, architecture, indent=0):
+        """
+        - Surrounds the lua line(s) contained within ``content`` with a premake "filter" and returns the result.
+        - A "filter" will affect all premake function calls after it's set. It's used to limit following project 
+          setup function call(s) to a certain scope. Here it is used to limit the calls in content to only apply
+          if the premake ``configuration`` and ``architecture`` matches the parameters in this function call.
+        """
+        lines = list(itertools.chain.from_iterable([cnt.splitlines() for cnt in content]))
+        return [
+            # Set new filter
+            f'{self.tab * indent}filter {{ "configurations:{configuration}", "architecture:{architecture}" }}',
+            # Emit content
+            *[f"{self.tab * indent}{self.tab}{line.strip()}" for line in list(filter(None, lines))],
+            # Clear active filter
+            f"{self.tab * indent}filter {{}}",
+        ]
+    
+    def _premake_filtered_fallback(self, content, configurations, architecture, indent=1):
+        """
+        - Uses filters that serve the same purpose than ``_premake_filtered()``
+        - This call will create an inverse filter on configurations. It will only apply if non of the 
+          ``configurations`` from this function call is present in premake. This is the fallback when a premake
```

Comment:
I am not sure about this functionality, other build systems do not implement it, but need the explicit settings to match (it is possible to define different configs with ``-s &:build_type=Debug -s build_type=Release``, but in any case, the config of the dependency is well defined, no fallback (that might cause issues in OS like Windows). Maybe we can start with something simpler without it?

Reply:
I have added this feature because I faced an issue several times in the past. 

For example if I create a configuration `Profile` in premake and try to apply the automatic `conan_setup()` function(s) it will end up with not applying the dependencys at all. With the implemented functionality it will automatically apply the `release` dependencys to all unknown configurations. This is an edge case that i have faced some times. 

I would prevere keeping it because I'm facing that edge case in several project. Not keeping that functionality would require me to manually use the lua variables and not be able to use the automatically generate functions. However if this functionality is not wanted I can remove it...

Reply:
With my latest commit this should no longer be relevant.